lib/galaxy/managers/workflows.py

```diff
@@ -1335,9 +1335,16 @@ def _workflow_to_dict_export(self, trans, stored=None, workflow=None, internal=F
         """
         annotation_str = ""
         tag_str = ""
+        annotation_owner = None
         if stored is not None:
             if stored.id:
-                annotation_str = self.get_item_annotation_str(trans.sa_session, trans.user, stored) or ""
+                # if the active user doesn't have an annotation on the workflow, default to the owner's annotation.
+                annotation_owner = stored.user
```

Comment:
Does `trans.user` ever make sense ? I don't think there's a way you can add annotations to a workflow if you're not the owner (through the API, of course the model allows it ...)

Reply:
I don't think there's currently a way to do this, and unless we imagine a workflow that has been published/shared and intentionally supports user-specific annotations diverging from the owner's, I don't think so.  I think we basically always just want whatever the owner put here, but then that'd mean to me that this shouldn't be a separate annotation object at all probably, and just merged into a simple field in the workflow/workflowStep as appropriate.  That's more than I'd want to put in a stable PR though.